test/framework/src/main/java/org/elasticsearch/test/ESIntegTestCase.java

```diff
@@ -1086,6 +1088,34 @@ public static List<PersistentTasksCustomMetadata.PersistentTask<?>> findTasks(Cl
         return tasks.tasks().stream().filter(t -> taskNames.contains(t.getTaskName())).toList();
     }
 
+    /**
+     * Waits for the health node to be assigned and returns the node
+     * that it is assigned to.
+     * Returns null if the health node is not assigned in due time.
+     */
+    @Nullable
+    public static DiscoveryNode waitAndGetHealthNode(InternalTestCluster internalCluster) {
+        DiscoveryNode[] healthNode = new DiscoveryNode[1];
+        try {
+            waitUntil(() -> {
+                ClusterState state = internalCluster.client()
+                    .admin()
+                    .cluster()
+                    .prepareState()
+                    .clear()
+                    .setMetadata(true)
+                    .setNodes(true)
+                    .get()
+                    .getState();
+                healthNode[0] = HealthNode.findHealthNode(state);
+                return healthNode[0] != null;
+            }, 15, TimeUnit.SECONDS);
```

Comment:
2 seconds seemed quite little to me so increased the timeout a bit to avoid potential flakiness

Reply:
You are probably right, +1.